audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,432 @@
+          const numClosingSteps =
+              getNumStepsFromOneDownTo_(linearThreshold, noisegate.alpha_);
+
+          const openIndex = numOpeningSteps + scriptProcessorLatency;
+          
+          const sourceBeneathThresholdIndex =
+              testFrames * timeToThreshold + scriptProcessorLatency;
+
+          let beforeOpenPhase =
+              channel.slice(scriptProcessorLatency, openIndex);
+          let openPhase = channel.slice(openIndex, sourceBeneathThresholdIndex);
+          let closePhase = channel.slice(
+              sourceBeneathThresholdIndex + numClosingSteps, channel.length);
+          let betweenOpenAndClosePhase =
+              channel.slice(sourceBeneathThresholdIndex,
+                            sourceBeneathThresholdIndex + numClosingSteps);
+
+          should(beforeOpenPhase, 'The samples before the gate opens')
+              .beConstantValueOf(0);
+          should(openPhase, 'The open phase').beConstantValueOf(1);
+          should(closePhase, 'The closed phase').beConstantValueOf(0);
+
+          // The relativeThreshold is set experimentally, and exists because
+          // of the imperfection of floating pointing multiplication. This
+          // test may fail for one sample if settings (e.g. alpha) change.
+          should(betweenOpenAndClosePhase, 'The samples before the gate closes')
+              .beCloseToArray(new Array(betweenOpenAndClosePhase.length)
+                                  .fill(beneathThresholdGain),
+                              {relativeThreshold : 1e-7});
+
+          // With current settings, the opening delay is 0.00049s
+          console.log('The opening delay is ' +
+                      toTime_(numOpeningSteps, sampleRate));
+
+          // With current settings, the closing delay is 0.0034s
+          console.log('The closing delay is ' +
+                      toTime_(numClosingSteps, sampleRate));
+
+        }).then(() => task.done());
+      });
+
+      // Verify that the noise gate computes an envelope for an audio signal
+      // that is on the same scale as web audio's gain node. If the gain of a
+      // signal is set to be beneath the threshold, then the signal should be
+      // muted. The gain of an oscillator is gradually decreased to some
+      // threshold T at timeToThreshold, and the output of the noise gate
+      // should be 0 after timeToThreshold and unaltered before timeToThreshold
+      // within an acceptable margin of error. This test quantifies
+      // delay on oscillators, and allows for the bandwidth of the filter to be
+      // experimentally determined in a somewhat realistic scenario.
+      // The bandwidth can be adjusted to minimize envelope delay while
+      // avoiding the gate rapidly changing between open and closed when the
+      // sample value is near the threshold.
+      audit.define('envelope-has-appropriate-alpha', (task, should) => {
```

Comment:
How do you set an alpha value? This test doesn't determine any alpha; it looks like you just picked something that you perhaps experimented with.  If so, you should say so in a comment.

Also, since the input is an oscillator you could analytically determine the response with a bit (a lot?) of work.